neuralmonkey/encoders/imagenet_encoder.py

```diff
@@ -46,18 +47,52 @@ class ImageNet(ModelPart, Attentive):
     WIDTH = 224
     HEIGHT = 224
 
-    # pylint: disable=too-many-arguments
+    # pylint: disable=too-many-arguments,too-many-locals
     def __init__(self,
                  name: str,
                  data_id: str,
                  network_type: str,
-                 output_layer: str,
+                 attention_layer: Optional[str],
+                 attention_state_size: int,
                  attention_type: Type=Attention,
                  fine_tune: bool=False,
+                 encoded_layer: Optional[str]=None,
                  load_checkpoint: Optional[str]=None,
                  save_checkpoint: Optional[str]=None) -> None:
+        """Initlize pre-trained ImageNet network.
+
+        Args:
+            name: Name of the model part (the ImageNet network, will be in its
+                scope, indepently on `name`)
+            data_id: Id of series with images (list of 3D numpy arrays)
+            network_type: Identifier of ImageNet network from TFSlim
+            attention_layer: String identifier of the convolutional map that
+                will be used for attention.
+            attention_state_size: Dimensionality of state projection in
+                attention computation.
+            attention_type: Type of attention.
+            fine_tune: Flag whether the network should be further trained with
+                the rest of the model.
+            encoder_layer: String id of the network layer that will be used as
+                input of a decoder. `None` means averaging the convolutional
+                maps.
+            load_checkpoint: Checkpoint file from which the pre-trained network
+                is loaded.
+            save_checkpoint: Checkpoint file where the encoder is saved after
+                the training. (Makes sense only if `fine_tune` is set to
+                `True`).
+        """
         ModelPart.__init__(self, name, save_checkpoint, load_checkpoint)
-        Attentive.__init__(self, attention_type)
+        Attentive.__init__(self, attention_type,
+                           attention_state_size=attention_state_size)
+
+        if attention_layer is None and attention_type is not None:
+            raise ValueError("Attention type is set, although "
```

Comment:
To je dobře, že to spadne. Když nemáš zadána, nad jakou vrstvou ImageNetové sítě se má dělát attention a chceš dělat attention (protože `attention_type` není `None`), tak to má spadnout.